Resources/public/js/Sonata.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * Sonata core object.
+ * Config, translation, and DOM helpers.
+ */
+;(function (window, $, Sonata) {
+    'use strict';
+
+    // Merge default config with data passed from the templates.
+    Sonata.config = $.extend({
+        confirmExit: true,
+        useSelect2: true,
+        useICheck: true,
+        useStickyForms: true
+    }, Sonata.config || {});
```

Comment:
Here we could have one config object per plugin, e.g.
```js
Sonata.config = {
  select2: {enabled: true, defaults: {foo: 'bar'}}
};
```
So users could easily override global config by doing:
```js
Sonata.config.select2.defaults.foo = 'bar';
```